src/issue-listing.jsx

```diff
@@ -0,0 +1,525 @@
+		@media (max-width: ${mediaQueries.max.medium}) {
+			{ content: "Created Time"; }
+		}
+	}
+
+	td:nth-of-type(4):before {
+		@media (max-width: ${mediaQueries.max.medium}) {
+			{ content: "Last Updated"; }
+		}
+	}
+`;
+Table.displayName = 'Table';
+
+const TableHeader = styled.thead`
+    th {
+        border-bottom: 0;
+        background: ${theme.backgroundAlt};
+        vertical-align: middle;
+    }
+
+    button {
+    	width: 100%;
+    	height: 100%;
+        padding: 0.75rem ${theme.gutter};
+        font-size: ${theme.smallBaseFont};
+
+        *[class*='button__Icon'] {
+            transform: scale(0.7);
+
+            @media (min-width: ${mediaQueries.min.medium}) {
+                transform: scale(1);
+            }
+        }
+    }
+`;
+TableHeader.displayName = 'TableHeader';
+
+export default class IssueListing extends Component {
+	   constructor(props) {
+        super(props);
+
+        this.state = {
+            issues: null,
+            sort: {
+           		column: 'created_at',
+           		direction: 'desc'
+            },
+            issuesLoaded: null
+        };
+
+        this.fetchIssues = this.fetchIssues.bind(this);
+        this.truncateByWord = this.truncateByWord.bind(this);
+        this.onSort = this.onSort.bind(this);
+        this.setArrow = this.setArrow.bind(this);
+        this.renderIssueTable = this.renderIssueTable.bind(this);
+        this.renderIssues = this.renderIssues.bind(this);
+        this.renderRepoName = this.renderRepoName.bind(this);
+        this.renderMobileSort = this.renderMobileSort.bind(this);
+        
+    }
+
+    fetchIssues() {
+    	const octokit = new Octokit({
+            auth: this.props.apiKey
+        });
+
+        octokit.issues.listForRepo({
+        	owner: this.props.selectedRepo.owner.login,
+        	repo: this.props.selectedRepo.name
+        }).then(({ data }) => {
+                this.setState({
+                    issues: data,
+                    issuesLoaded: true
+                });
+                this.onSort(null, this.state.sort.column, false)
+                this.setArrow(this.state.sort.column)
+            }).catch(err => {
+            	this.setState({
+                    issues: [],
+                    issuesLoaded: true
+                });
+            });
+    }
+
+    componentDidMount() {
+        const column = window.sessionStorage.getItem("sort_column") || 'created_at';
+        const direction = window.sessionStorage.getItem("sort_direction") || 'desc';
+
+        this.setState({
+            sort: {
+                column: column,
+                direction: direction
+            }
+        });
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.selectedRepo !== this.props.selectedRepo) {
+        	if (this.props.selectedRepo) {
+        		this.setState({ issuesLoaded: false });
+            	this.fetchIssues();
+        	}
+        }
+    }
+
+    onSort(e, column, sortDirection=true) {
```

Comment:
I'd try to pull a few different functions out of this one in order to name them, to make it a bit easier for a reader to see what's going on in here. 

Potentially the title logic
```
        if (column === 'title') {
            const titleA = a.title.toUpperCase();
            const titleB = b.title.toUpperCase(); 

            if (titleA < titleB) {
                return -1;
            }
            if (titleA > titleB) {
                return 1;
            }
            return 0;
```
the Avatar URL logic 
```
                const assigneeA = a.assignee && a.assignee.login.toUpperCase();
                const assigneeB = b.assignee && b.assignee.login.toUpperCase(); 
                if (assigneeA < assigneeB) {
                    return -1;
                }
                if (assigneeA > assigneeB) {
                    return 1;
                }
                return 0;
```

Might even find another one or two in there.  😄